src/components/Tabs/TabsBar/TabsBar.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import bem from '../../../utils/bem';
+import { TabItem, TabItemProps } from '../TabItem';
+import styles from './TabsBar.scss';
+
+interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+    /** Id of currently active tab. This will overwrite isActive on the children */
+    activeTabId?: string | number;
+    /** The tabs */
+    children: React.ReactElement<TabItemProps> | React.ReactElement<TabItemProps>[];
+    /** Callback function, fired when switching tabs by clicking. This will overwrite onSelect on children, if defined */
+    onSelect?: (selectedTabID: number | string) => void;
+    /** to render the tabs so they cover the full width of the available space */
+    isBlock?: boolean;
+}
+
+const { block } = bem('TabsBar', styles);
+
+const TabsBar: React.FC<Props> = props => {
+    const { activeTabId, children, onSelect, isBlock, ...rest } = props;
+
+    const extendWithProps = (tab: Props['children']) => {
+        if (React.isValidElement(tab) && tab.type === TabItem) {
+            const { tabId } = tab.props;
+            const extendProps: {
+                isBlock?: boolean;
+                isActive?: boolean;
+                onSelect?: (selectedTabID: number | string) => void;
```

Comment:
This TypeScript assigning a bit confused me, can we move it to the top of the file?

Reply:
Tiny detail but should this perhaps be `selectedTabId` (lowercase `d`) to be consistent with other places that mention `tabId`?